kornia/geometry/nerf/nerf.py

```diff
@@ -0,0 +1,340 @@
+import os
+import unittest
+from typing import Tuple
+
+import cv2
+import numpy as np
+import torch
+import torch.nn as nn
+from torch.optim.lr_scheduler import MultiStepLR
+
+from kornia.geometry.nerf.from_nerfmm import render_novel_view, TinyNerf, train_one_epoch
+from kornia.geometry.nerf.nerfmm.models.intrinsics import LearnFocal
+from kornia.geometry.nerf.nerfmm.models.poses import LearnPose
+from kornia.geometry.nerf.nerfmm.utils.training_utils import mse2psnr
+from kornia.utils import image_list_to_tensor
+
+
+class RayParameters:
+    def __init__(self):
+        self.NEAR, self.FAR = 0.0, 1.0  # ndc near far
+        self.N_SAMPLE = 128  # samples per ray
+        self.POS_ENC_FREQ = 10  # positional encoding freq for location
+        self.DIR_ENC_FREQ = 4  # positional encoding freq for direction
+
+
+class LearnFocals(nn.Module):
```

Comment:
wondering how we could leverage here the `PinholeCamera` data structure
https://kornia.readthedocs.io/en/latest/geometry.camera.pinhole.html#kornia.geometry.camera.pinhole.PinholeCamera

Reply:
Should be possible to use this DS as a unified container for the estimated camera parameters